src/jit/flowgraph.cpp

```diff
@@ -8034,6 +8034,7 @@ GenTree* Compiler::fgCreateMonitorTree(unsigned lvaMonAcquired, unsigned lvaThis
             GenTree* temp = fgInsertCommaFormTemp(&retNode->gtOp.gtOp1, info.compMethodInfo->args.retTypeClass);
 
             GenTree* lclVar                 = retNode->gtOp.gtOp1->gtOp.gtOp2;
+            lclVar->gtFlags |= (retExpr->gtFlags & GTF_DONT_CSE);
```

Comment:
It might be worth a comment describing this case.
I think that it's probably generally prudent to propagate the `GTF_DONT_CSE`, but you could mention that the return can't handle all of the trees that could be on the right-hand-side of an assignment, especially in the case of a struct. In fact, it might be sufficient to propagate the flag only in the case that it's a struct, though perhaps it's best to be conservative.

Reply:
Added comment, thanks.